vasyl_varyshchuk/Task_2/pass_arg_decorator.py

```diff
@@ -0,0 +1,39 @@
+"""
+This module implements a retry decorator with ability to pass a parameter into decorator
+"""
+
+import random
+
+
+def decorator(retries=None):
+
+    def my_decorator(func):
+
+        def func_wrapper():
+            if retries is not None:
+                for i in range(1, retries+1):
+                    try:
+                        func()
+                        break
+                    except Exception:
+                        print('Restart => {}'.format(i))
+            else:
```

Comment:
What is the reason of this `else` statement?
Do we really need it?

Reply:
I think so. In the case we do not pass the parameter to the decorator(None value by default), the function will not run. Are there any ways to overcome it?

Reply:
Hm..

Reply:
Thanks for your advice.